into/backends/iopro.py

```diff
@@ -0,0 +1,159 @@
+    My understanding is that odbc.ini is only required for pre-defined
+      data-source names (DSNs).
+    No DSNs are used here because we extract all necessary data from
+      the sqlalchemy engine.
+    """
+    url = engine.url
+    connect_args = url.translate_connect_args()
+
+    server = connect_args.get('host','')
+    uid = connect_args.get('username','')
+    password = connect_args.get('password','')
+    database = connect_args.get('database','')
+    port = connect_args.get('port','')
+    option = connect_args.get('option','')
+
+    driver = dialect = url.get_backend_name()
+    
+    if driver == "mysql":
+        #option=3 is required by mysql
+        option = "3"
+        dialect = "mysql+pyodbc"
+    elif driver == "mssql":
+        dialect = "mssql+pyodbc"
+    else:
+        raise NotImplementedError("sqlalchemy does not support the {} database "\
+                                  "with the pyodbc dialect yet.".format(driver))
+        
+    #Note: It appears to be OK to leave parameters empty. Default
+    #  values will be filled in by the odbc driver.
+    connection_string = "DRIVER={{{}}};SERVER={};PORT={};DATABASE={};UID={};"\
+                        "PASSWORD={};OPTION={};".format(driver, server, port,
+                                                        database, uid, password,
+                                                        option)
+
+    def getconn():
+        #If the odbc connection cannot be made, raising the
+        #  NotImplementedError exception should make the convert
+        #  dispatch manager fall back to a working conversion path.
+        try:
+            iopro_db_conn = iopro.pyodbc.connect(connection_string)
+        except Exception as e:
+            raise NotImplementedError("iopro odbc connection failed. "
+                                      "Falling back to defaults. "
+                                      "Original exception: %s" % (e))
+        return iopro_db_conn
+
+    new_sa_engine = sa.create_engine("{}://".format(dialect), creator=getconn)
+
+    return new_sa_engine
+
+    
+#The cost=26.0 comes from adding up all of the costs from the original
+#  convert() path and dividing by 15 (15x is the new speedup)
+@convert.register(np.ndarray, sa.sql.Select, cost=26.0)
+def iopro_sqlselect_to_ndarray(sqlselect, **kwargs):
+    """Return a numpy struct-array from a sql.Select
+
+    Parameters
+    ----------
+    sqlselect - a sqlalchemy sql.Select object
+
+    Return
+    ------
+    A numpy struct array with the sql results
+
+    Notes
+    -----
+    In testing, using fetchsarray() on mysql is about
+    15x faster than naive methods at the 10,000 row
+    range.
+    """
+
+    #Remake the database engine so that iopro can use fetchsarray()
+    orig_engine = sqlselect.bind
+    iopro_db_engine = get_iopro_db_engine(orig_engine)
+
+    conn = iopro_db_engine.raw_connection()
+    cursor = conn.cursor()
+    cursor.execute(str(sqlselect))
```

Comment:
In the case of parametrized queries consider using https://github.com/ContinuumIO/blaze/blob/master/blaze/compute/utils.py#L11